gdal/ogr/ogrsf_frmts/mysql/ogrmysqltablelayer.cpp

```diff
@@ -383,9 +383,13 @@ OGRFeatureDefn *OGRMySQLTableLayer::ReadTableDefinition( const char *pszTable )
         poDefn->SetGeomType( wkbUnknown );
         poDefn->GetGeomFieldDefn(0)->SetName( pszGeomColumn );
 
-        osCommand = "SELECT type, coord_dimension FROM geometry_columns WHERE f_table_name='";
-        osCommand += pszTable;
-        osCommand += "'";
+        if( poDS->GetMajorVersion() < 8 || poDS->IsMariaDB() )
+            osCommand.Printf("SELECT type, coord_dimension FROM geometry_columns WHERE f_table_name='%s'",
+                             pszTable);
+
+        else
+            osCommand.Printf("SELECT GEOMETRY_TYPE_NAME FROM INFORMATION_SCHEMA.ST_GEOMETRY_COLUMNS "
```

Comment:
The deprecated selected statement above includes the coord_dimension whereas this select statement does not. Does support need to be included for 3 dimensional coordinate space?

Reply:
Also line 409 `if( papszRow[1] != nullptr && atoi(papszRow[1]) == 3 )` may be accessing an array out of bounds since that column was not selected?

Reply:
In my understandings,  MySQL 8  has only 2D functions, and integrated ST_GEOMETRY_COLUMNS table does not have a column as same meaning as coord_dimensions. I saw here: https://dev.mysql.com/doc/refman/8.0/en/st-geometry-columns-table.html
@YoshiakiYamasaki could  you help about a 3D data parameter?

Reply:
Implementation is under consideration in the future, but 3D is not implemented now. Currently MySQL 8.0 only supports 2D.

FYI:
Spatial Support in MySQL
https://www.slideshare.net/NorvaldRyeng/spatial-support-in-mysql/5

* This document made by derector of MySQL GIS development team.

Reply:
Now line 409  is executed only when MySQL<8 and MariaDB.